lib/streamlit/report_session.py

```diff
@@ -417,6 +399,34 @@ def _enqueue_report_finished_message(self, status):
         msg.report_finished = status
         self.enqueue(msg)
 
+    def handle_git_information_request(self):
+        msg = ForwardMsg()
+
+        try:
+            from streamlit.git_util import GitRepo
+
+            self._repo = GitRepo(self._report.script_path)
```

Comment:
nit: It seems a bit weird that `self._repo` gets set here when it isn't accessed anywhere outside of the method (except in `self.get_deploy_params`, which we seem to be deprecating). May want to change this to just `_repo` (or something else since there's also a `repo` variable name)